code-challenges.test.js

```diff
@@ -27,25 +41,119 @@ const colors2 = [
 ]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-// b) Create the function that makes the test pass.
+describe("colorShuffle", () => {
+  it ("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+    expect(colorShuffle(colors1)).toEqual(["yellow", "blue", "pink", "green"]) 
+    expect(colorShuffle(colors2)).toEqual(["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"])
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+//   colorShuffle
+//     ✕ takes in an array, removes the first item from the array and shuffles the remaining content (1 ms)
+
+//   ● colorShuffle › takes in an array, removes the first item from the array and shuffles the remaining content
 
+//     ReferenceError: colorShuffle is not defined
+
+// b) Create the function that makes the test pass.  
+const colorShuffle = (colors) => {
+  let shuffledColors = [...colors]
+  for(let i=0; i<colors.length; i++){
+    colors.shift(i)
+    const shuffle = Math.floor(Math.random () * (i+1))
+    var temp = colors[i]
+    colors[i] = shuffledColors
+    shuffledColors = temp
+    shuffledColors(shuffle)
+  }
+} 
+
+console.log(colorShuffle(colors1))
+console.log(colorShuffle(colors2))
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
+//Psuedocode: 
+// Input: { upVotes: 13, downVotes: 2 }
+//        { upVotes: 2, downVotes: 33 }
+// Output: 11 & -31
+// write a test that will include expect and to euqal statements for the variables provided
+// run the test and see it fail it should be a not defined error
+// create a function 
+// console log the test and variables
+// test should pass
+
 // a) Create a test with expect statements for each of the variables provided.
 
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
+describe("totalVotes", () => {
+  it ("takes in an object that contains up votes and down votes and returns the end tally", () => {
+    expect(totalVotes(votes1)).toEqual([11]) 
+    expect(totalVotes(votes2)).toEqual([-31])
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+//   totalVotes
+//   ✕ takes in an object that contains up votes and down votes and returns the end tally
+
+// ● totalVotes › takes in an object that contains up votes and down votes and returns the end tally
+
+//   ReferenceError: totalVotes is not defined
+
 // b) Create the function that makes the test pass.
 
```

Comment:
This is SO close! This will not access the object because it is not inside of a class the function is independent